server/service/software_titles.go

```diff
@@ -117,8 +128,26 @@ func (svc *Service) SoftwareTitleByID(ctx context.Context, id uint) (*fleet.Soft
 		return nil, err
 	}
 
-	software, err := svc.ds.SoftwareTitleByID(ctx, id)
+	vc, ok := viewer.FromContext(ctx)
+	if !ok {
+		return nil, fleet.ErrNoContext
+	}
+
```

Comment:
I like this idea, will explore if there are any hidden gotchas, and try to go with it.

Reply:
@gillespi314 yeah that would probably be equivalent, not sure which is simpler - the nice thing here is that you don't need a different datastore method, but on the downside you have this slightly hacky way if re-calling `SoftwareTitleByID` with a "virtual global admin" to check if the software exists or not (we do use this "virtual user" approach elsewhere too).

If you collected the team ids first, you'd have to authorize for each team and ensure the user has access to at least one, but then I think you'd still need to pass the user to the datastore layer and add the dynamic where clause on its allowed teams, so that it sums up the correct counts that the user has the right to see (this is similar to what we do in other places where we filter the sql dynamically on the user's teams).

And to differentiate between a software that does not exist and one that the user doesn't have access to, you'd still need to do another check, I believe? Ie. a software may exist but have no counts for the teams the user has access to, so this is "forbidden", but if the software does not exist it would look the same at the authz check, so you'd need to verify its existence to return a 404? Unless a software cannot exist without a host count > 0 for at least one team (I don't recall if counts are updated on a cron job).

Reply:
w00t! you saved me a good chunk of exploration. Thank you!!

Reply:
(note that it depends how product wants the counts to appear, I'm assuming it should show the counts that the user is allowed to see - if user can see team 1 and 2 but not 3, it will show the sum of this software on hosts in teams 1 and 2 and will ignore any counts in team 3), and this is what the current approach would do)